// back/Model/evento.js
export default class Evento {
    //atributos Privados

    #nome
    #data
    #horario
    #local
    #preco
    #quantidade
    #descricao

    constructor(nome, data, horario, local, preco, quantidade, descricao) {
        this.#nome = nome;
        this.#data = data;
        this.#horario = horario;
        this.#local = local;
        this.#preco = preco;
        this.#quantidade = quantidade;
        this.descricao = descricao;

    }

    get nome(){
        return this.#nome;
    }

    set nome(newNome){
        this.#nome = newNome;
    }

    get data(){
        return this.#data;
    }

    set data(newData){
        this.#data = newData;
    }

    get horario(){
        return this.#horario;
    }

    set horario(newHorario){
        this.#horario = newHorario;
    }

    get local(){
        return this.#local;
    }

    set local(newLocal){
        this.#local = newLocal;
    }

    get preco(){
        return this.#preco;
    }

    set preco(newPreco){
        this.#preco = newPreco;
    }

    get quantidade(){
        return this.#quantidade;
    }

    set quantidade(newQuantidade){
        this.#quantidade = newQuantidade;
    }

    get descricao(){
        return this.#descricao;
    }

    set descricao(newDescricao){
        this.#descricao = newDescricao;
    }



    editorText() {
        return `Nome: ${this.nome} \n
        Data: ${this.data} \n
        Horario: ${this.horario} \n
        Local: ${this.local} \n
        Preco: ${this.preco} \n
        Quantidade: ${this.quantidade} \n
        Descricao: ${this.descricao}`
    }

}
